### DISCUSSION TITLE
Difficult to map long ways

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#3646] [Feedback]
#### Use case
I often plan my walking routes in advance, to methodically cover a whole area, It's hard to map a long way this way, because it's hard to keep track of my answer for it without walking the whole length; this often means taking a long detour, which I don't want to do (because I planned my route in advance). So, I'm often unable to answer the quests for long ways. This is frustrating since I do actually cover the entire way, just not all at once.

#### Proposed solution
When answering a quest for an extremely long way, split the quest into smaller segments. Do not upload my answers until I have completed all the segments. If I answer all segments with the same answer, tag the way with that answer. If not, prompt me to split the way.

It could be streamlined by making other segments a yes/no answer ("Is this section also ${what I picked for the first segment}?"

#### Current workaround
I can approximate the above behavior by using manual upload, answering the quest before I can see the entire way, and then undoing my answer if I need to split the way later. But I don't do this, because it's easy to screw up and forget to check later. :grimacing: So that's why I'd like if the app could help me remember, here.

---

See also https://github.com/streetcomplete/StreetComplete/pull/3280  
/cc @dbdean

**Comment by matkoniecz** [comment]
In similar cases I often simply use way splitting, and try to split in places where it will be anyway split in the future.

**Reply by smichel17** [reply]
I guess this works… I don't really like it, because I don't always know if/where it's likely to be split in the future. At least, not without walking it, which is what I'm trying to avoid in the first place…

**Comment by dbdean** [comment]
I map to exercise, and I'm not a completionist (at least not in one trip) so I tend to just follow the quests, and don't mind if long quest pull me in a different direction. It can be a challenge if it's too long (to walk) though. And if I miss a long quest, then I can always walk that way again next time.

I don't mind the idea of splitting ways preemptively to get answers in principle, but I would worry that it may be a long time before I come back. When would SC decided that I aren't going to answer for the rest of the way and decide to upload what I have?

I do wonder if SC could automatically *merge* ways when it ends up with a situation when every tag on two joining ways is identical. But I imagine with relationships there would be a lot of cases where that would not be a good idea.

**Reply by matkoniecz** [reply]
> I do wonder if SC could automatically merge ways when it ends up with a situation when every tag on two joining ways is identical.

Requires processing relations, and fetching all relations in a given area turned out to be tragic for performance.

> I don't mind the idea of splitting ways preemptively to get answers in principle, but I would worry that it may be a long time before I come back.

In case of splitting SC would upload split way (for now both not answered). After solving one of them second will remain untagged and wait for mappper

> When would SC decided that I aren't going to answer for the rest of the way and decide to upload what I have?

I am not getting the problem described here. What exactly may go wrong?

**Reply by dbdean** [reply]
I believe that @smichel17 was indicating that SC would pre-emptively split the ways somehow (locally), then only upload when the full way had all 'segments' answered, and split the way on upload only if there are differences in the answers.

So, in this case, when does SC upload my work if I don't end up answering for every 'segment'?

**Reply by matkoniecz** [reply]
I assumed that with this proposal such ways would "obviously" split and upload ways which were not finished for a long time, but it appears to be not specified.

**Reply by smichel17** [reply]
I left it unspecified because I'm not sure what makes the most sense, nor do I care. Whatever works best.

**Comment by timothywashere** [comment]
RE the OP, I had been wondering along the same lines.
I usually end up splitting a way just to make it manageable. When I join a way halfway along I often split it so that I can answer the part I am walking. I'm not sure if this is OSM good practice or not.

When an area has lots of terraces of house numbers needed, I try to walk in a pattern so I get the house numbers at each end of multiple terraces and then later on I might walk past the other end of the terraces and get the numbers at the other end. This is a great way of doing numbers but makes it hard to do surfaces because I can't remember what one end was like when I do the other.

So I agree, it would be a great feature if you could answer for one end of the street and SC would keep track of that answer and only if you answer the same at the other end it would allow you to upload it.  But I think it would be hard to implement.

Ideally it would offer a prompt if it thinks you have not surveyed the whole way, offering you the chance to split the way, cache the answer, ignore the warning or cancel the operation. But since SC doesn't track you when the screen is off, it might not know if you've walked the whole way or not.

**Reply by lynxdingo** [reply]
As a current work-around:

You could always create a note within SC which is then available for all other mappers to see. Sometimes the note will be resolved quickly, other times, you will have to resolve it yourself with an editor of your choosing (like iD, JOSM, Vespucci).

**Comment by peternewman** [comment]
As a more general note around long ways, it would be nice if the repeated tags had some indication, e.g. > or >> to show it continues or something. I often found on long ones (similar to your issues) mapping and managing them was hard, but also a route off to the side could either be a few tens of a metres or a few km and there's no way to tell. Likewise whether I've joined a way at the middle or the end, apart from selecting the way, then potentially having to zoom back in to work out if I'm actually right at the end of it.

**Comment by smichel17** [comment]
All the comments here have given me another idea. Hypothesis: Maybe the main issue is not having the walk the long way, but the **inability to plan** for these detours, because I don't know whether I'll have to take a detour until I **open the quest.**

> I map to exercise, and I'm not a completionist (at least not in one trip) so I tend to just follow the quests, and don't mind if long quest pull me in a different direction

Me too, somewhat. My rational brain mainly wants to walk for excercise. But my monkey brain cares much more about completionism and making the number in the top left go up.

I usually plan my route (at least roughly) in advance, based on where the quest pins are. I'd either go for being completionist or for making some progress in a dense area. Long ways are the only quest where I **can't rely on the map pins to plan my route**. I might open the quest and discover I can't answer it without taking a long detour.

What if there were some indication that "this quest is for a long way" before you open it? Ideally the indicator would also give a hint about where the way is (e.g. different for end pins vs start pins). This would make discovering a quest for a long way less annoying, without getting into the issue of what to do with a partially-answeed quest.

**Reply by peternewman** [reply]
> What if there were some indication that "this quest is for a long way" before you open it? Ideally the indicator would also give a hint about where the way is (e.g. different for end pins vs start pins). This would make discovering a quest for a long way less annoying, without getting into the issue of what to do with a partially-answeed quest.

Is there an echo here @smichel17 ? https://github.com/streetcomplete/StreetComplete/discussions/3646#discussioncomment-1994694 :laughing: 

In terms of the detour, when you get to a junction you could also see at a glance whether it's go the wrong way and double back for a few minutes, or whether you'd need to do a huge detour to get the whole way.

**Reply by smichel17** [reply]
Yes, inspired by your comment (and others). But yours seemed focused on after you tapped on the quest, rather than before.

**Reply by peternewman** [reply]
No, very poorly described, but I sort of imagined some additional decoration to the existing quest pin when it's one of many pins for a quest (especially for the long ones, but might also make sense for things like speed where they are at the ends of the way.

**Comment by RubenKelevra** [comment]
I just wanted to start an a discussion about this too.


Especially with the road surface quality (smoothness) I find it difficult to answer it for large sections at a time.


So I tend to use the split way function a lot more often than previously, to cut the way in "overlook-able" sections.

I was wondering why we don't automate this, as it takes quite long to do this manually.

SC could show the quest for just 50m strips at a time regardless how the way length below is. If a user is answering all sections it can concatenate the answers and just modify the whole section.

This would allow us to remove the max length filter which hides quests on very long ways, which id blocking users on cutting it down and answering it. 

This obviously need some "caching" of the answers to give the user time to answer them all. So if the user tabs out of SC SC could start uploading it - or if the users leaves the general area we could do that too.

**Comment by InsertUser** [comment]
I've had this concern as well. My Ideal solution would be for SC to ask me how far I can see when answering questions about long or curved roads. That way if I give consistent answers along the overlapping view areas it can upload in one go, otherwise it can split the way at the edges of the viewed areas?
